Overview openapi (#209)
SAP/cloud-sdk: docs/java/features/rest/overview.mdx

hide_title: false
hide_table_of_contents: false
sidebar_label: Overview
description: SAP Cloud SDK offers type-safe client generators for OpenAPI services
keywords:
  - sap
  - cloud
  - sdk
  - cloud native
  - cloud sdk
  - sap cloud sdk
  - openapi
image:
---

:::info Released as generally available
OpenAPI code generator and type-safe client are released as generally available and ready for productive use.
:::

## OpenAPI Based Type-Safe Client for RESTful Services in Java

> [OpenAPI](https://www.openapis.org/) standard is a popular format to define RESTful APIs. Together with [OData](../odata/overview.mdx) it is one of the key API definition formats used at SAP. The SAP Cloud SDK provides convenient tooling to make the consumption of OpenAPI services easy and convenient. We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK. To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available through [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*). Another way to get a client library is to generate it based on API specifications available at [SAP API Business Hub](https://api.sap.com/). If you encounter any problems or errors with OpenAPI tooling give us feedback via our [support channels](../../../overview/support.mdx).

Reviewer: "SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK" - can you add a link here?
Maybe you want to make even clearer that the pre-generated libraries are an additional convenience, but not necessary for developers to benefit from the generator.

Author: @sacnl, I think it has enough links. A customers can read about our other features by navigating through the docs. There they are explained with enough details.

### Pregenerated Type-Safe OpenAPI Client

We ship pre-generated type-safe OpenAPI client libraries as modules in collaboration with popular SAP services available on SAP Business Technology Platform and beyond.

Depending on the scope, modules could be available only internally within SAP or publicly released.
The [SAP Workflow service](clients/scp-workflow-rest-api) is an example of a publicly released API module.

## Why Using SAP Cloud SDK for Java With OpenAPI Services?

> - You'll benefit from a type-safe, less code boilerplate, and better developer experience.

Reviewer: ```suggestion
- You'll benefit from type-safety, less code boilerplate, and better developer experience.
```

hide_title: false
hide_table_of_contents: false
sidebar_label: Overview
description: SAP Cloud SDK offers type-safe client generators for OpenAPI services
keywords:
  - sap
  - cloud
  - sdk
  - cloud native
  - cloud sdk
  - sap cloud sdk
  - openapi
image:
---

:::tip Released as generally available

OpenAPI code generator and type-safe client are released as generally available and ready for productive use.

:::

## OpenAPI Based Type-Safe Client for RESTful Services in Java

[OpenAPI](https://www.openapis.org/) standard is a popular format to define RESTful APIs.
Together with [OData](../odata/overview.mdx) it is one of the key API definition formats used at SAP.
The SAP Cloud SDK provides convenient tooling to make the consumption of OpenAPI services easy and convenient.

> We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK.

Reviewer: ```suggestion
We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK.
You can use that generator to create a client for an OpenAPI-based API of your choice.
```

Author: Is it smth special to our policies: "of your choice"? I meet it often but from a meaning perspective, it seems rather verbose if not to say redundant. No hard feelings, I'll accept it. I personally stumble over those and it makes docs more difficult to read for me.

> To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available through [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*).

Reviewer: ```suggestion
On top, for additional convenience, to speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available through [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*).
```

Author: Here I'd leave either my version or yours. If yous to be taken it's better to remove "to speed up things for developers and reduce maintenance" as yours already implies it. The main goal of this abstract is still emphasizing clients at the end.

> Another way to get a client library is to generate it based on API specifications available at [SAP API Business Hub](https://api.sap.com/).

Reviewer: ```suggestion
Another way to get a client library is to generate it yourself based on API specifications available at [SAP API Business Hub](https://api.sap.com/).
```

Author: This also seems self-explanatory to me. Who else would generate it for you? But if you've seen our customers missing the point in such situation I'll accept it.
